File: Go/BubbleSort/src/bubblesort.go
package main
import "fmt"

func main(){
	var arr = [10]int{1,45,6,12,3, 20, 16, 11, 4,2}
	var arr_len int = len(arr)

	var i, j int
	for i = 0; i<arr_len-1; i++{
		for j = 0; j<arr_len-i-1; j++{
			if (arr[j] > arr[j+1]){
				var temp int = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}


	fmt.Printf("Array (n=%d): ", arr_len);
	var n int
	for n = 0; n<arr_len; n++{
		if (n !=9){
			fmt.Printf("%d, ", arr[n])
		}else{
			fmt.Printf("%d.", arr[n])
		}
	}
}

File: Rust/BubbleSort/bubblesort.rs
fn main(){
	let mut arr:[u32;10] = [1,45,6,12,3, 20, 16, 11, 4,2];
	let arr_len = arr.len();

	for i in 0..arr_len-1{
		for j in 0..arr_len-i-1{
			if arr[j] > arr[j+1]{
				let temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
	}

	println!("Array (n={}): ", arr_len);
	for n in 0..arr_len{
		println!("{}", arr[n]);
	}
}
